expression/builtin_cast_vec.go

```diff
@@ -836,7 +836,7 @@ func (b *builtinCastRealAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result
 				if types.ErrOverflow.Equal(err) {
 					warnErr := types.ErrTruncatedWrongVal.GenWithStackByArgs("DECIMAL", b.args[0])
 					err = b.ctx.GetSessionVars().StmtCtx.HandleOverflow(err, warnErr)
-				} else if types.ErrTruncated.Equal(err) {
+				} else if strings.Contains(err.Error(), "Truncated incorrect") {
```

Comment:
Why do we do string matching for identifying an error? It is quite brittle.

Reply:
Agree, sorry, i can't find more elegant methods to handle this, please do me a favor about this point

Reply:
1. Can we use types.ErrTruncatedWrongVal.Equal(err)?
2. Non-vec implement also need modify.
3. Are you sure we need modify CastRealAsDecimal, instead of CastStringAsDecimal? 
4. Is there any related test? I didn't find the test cover the case.. Or maybe I ignore the test?